src/lib/components/table/table-column-settings/dialog/table-settings-dialog.component.ts

```diff
@@ -32,7 +34,57 @@ export class TableSettingsDialogComponent implements OnInit
      */
     public ngOnInit():void
     {
-        this._columns = this.data.columns;
         this._selectedColumns = this.data.selectedColumns;
+        this._sort(this.data.columns);
+    }
+
+
+    public _onDrop(event:CdkDragDrop<Array<MatColumnDef>>):void
+    {
+        moveItemInArray(this._columns, event.previousIndex, event.currentIndex);
+
+        this._sort(this._columns);
+    }
+
+    public _updateSelectedList():{ sortedList:Array<string>, columns:Array<MatColumnDef> }
+    {
+        let sortedList:Array<string> = [];
+
+        this._columns.forEach((col:MatColumnDef) =>
+        {
+            let index:number = this._selectedColumns.indexOf(col.name);
+
+            if(index > -1)
+            {
+                sortedList.push(col.name);
+            }
+        });
+
+        return {
+            sortedList: sortedList,
+            columns:    this._columns
+        };
+    }
+
+    public _sort(cols:Array<MatColumnDef>):void
+    {
+        let selectedList:Array<MatColumnDef> = [];
+        let unselectedList:Array<MatColumnDef> = [];
+
+        cols.forEach((col:MatColumnDef) =>
+        {
+            let index:number = this._selectedColumns.indexOf(col.name);
+
+            if(index > -1)
+            {
+                selectedList.push(col);
+            }
+            else
+            {
+                unselectedList.push(col);
+            }
+        });
+
+        this._columns = selectedList.concat(unselectedList);
```

Comment:
Lieber die sortierte Liste zurückgeben
```suggestion
    public _sort(cols:Array<MatColumnDef>):Array<MatColumnDef>
    {
        let selectedList:Array<MatColumnDef> = [];
        let unselectedList:Array<MatColumnDef> = [];

        cols.forEach((col:MatColumnDef) =>
        {
            let index:number = this._selectedColumns.indexOf(col.name);

            if(index > -1)
            {
                selectedList.push(col);
            }
            else
            {
                unselectedList.push(col);
            }
        });

        return selectedList.concat(unselectedList);
```